exonum/src/sandbox/consensus/block_request.rs

```diff
@@ -304,6 +304,85 @@ fn handle_block_response_with_incorrect_tx() {
     // an incorrect tx, node should panic.
 }
 
+/// - should **NOT** process block if tx is incorrect
+/// idea of test is:
+/// - getting Status from other node with later height, send BlockRequest to this node
+/// - receive incorrect tx
+/// - receive BlockResponse with known tx A
+/// - Figure out that block is incorrect
+/// - Node should panic because of committed block with incorrect tx.
+#[test]
+#[should_panic(expected = "Received a block with transaction known as invalid")]
+fn handle_block_response_with_known_incorrect_tx() {
```

Comment:
IMO, it could make sense to unite the test logic with `handle_block_response_with_incorrect_tx` by making it parametric over whether the transaction is received before the block response:

```rust
fn test_handle_block_response_with_incorrect_tx(known_before_block: bool) {
    // test logic here
}

#[test]
fn handle_block_response_with_incorrect_tx() {
    test_handle_block_response_with_incorrect_tx(false);
}

#[test]
fn handle_block_response_with_known incorrect_tx() {
    test_handle_block_response_with_incorrect_tx(true);
}
```